source/grid/tria.cc

```diff
@@ -12520,6 +12520,34 @@ void Triangulation<dim, spacedim>::copy_triangulation(
   if (other_tria.policy)
     this->policy = other_tria.policy->clone();
 
+  // periodic faces
+  this->periodic_face_pairs_level_0.reserve(
+    other_tria.periodic_face_pairs_level_0.size());
+
+  for (auto entry : other_tria.periodic_face_pairs_level_0)
+    {
+      entry.cell[0] =
+        cell_iterator(this, entry.cell[0]->level(), entry.cell[0]->index());
+      entry.cell[1] =
+        cell_iterator(this, entry.cell[1]->level(), entry.cell[1]->index());
+      periodic_face_pairs_level_0.emplace_back(entry);
+    }
+
+  for (auto [first_cell_, second_cell_and_orientation] :
+       other_tria.periodic_face_map)
+    {
+      auto first_cell  = first_cell_; // make copy since key is const
```

Comment:
This illustrates the point I'm trying to make. `first_cell_` here is a `const` reference to an object in the other triangulation, but this is not at all obvious to a reader. If that had been the same in the code above, you would have modified the *other* triangulation.